src/main/scala/net/snowflake/spark/snowflake/io/SnowflakeResultSetRDD.scala

```diff
@@ -125,9 +127,19 @@ case class ResultIterator[T: ClassTag](
     }
   }
 
+  context.addTaskCompletionListener[Unit](_ => closeResultSet())
```

Comment:
Can we avoid introduce `context` by using `TaskContext.get()`?
`TaskContext.get().addTaskCompletionListener[Unit](_ => closeResultSet())`